x-pack/elastic-agent/pkg/agent/operation/operator.go

```diff
@@ -244,18 +252,36 @@ func (o *Operator) getApp(p Descriptor) (Application, error) {
 	}
 
 	// TODO: (michal) join args into more compact options version
-	a, err := app.NewApplication(
-		o.bgContext,
-		p.ID(),
-		p.BinaryName(),
-		o.pipelineID,
-		o.config.LoggingConfig.Level.String(),
-		specifier,
-		o.srv,
-		o.config,
-		o.logger,
-		o.reporter,
-		o.monitor)
+	var a Application
+	var err error
+	if p.ServicePort() == 0 {
+		a, err = process.NewApplication(
+			o.bgContext,
+			p.ID(),
+			p.BinaryName(),
+			o.pipelineID,
+			o.config.LoggingConfig.Level.String(),
+			specifier,
+			o.srv,
+			o.config,
+			o.logger,
+			o.reporter,
+			o.monitor)
+	} else {
+		a, err = service.NewApplication(
+			o.bgContext,
+			p.ID(),
+			p.BinaryName(),
+			o.pipelineID,
+			o.config.LoggingConfig.Level.String(),
+			p.ServicePort(),
```

Comment:
Too me a while to find the diff. Any change to have a comment around line 257 on why if ServicePort==0 it starts a process and not a service?

Reply:
Added a comment toi explain.